### PR TITLE
Add option to choose whether #PasswordDigest or #PasswordText is used…

### PR DESCRIPTION
**Pull request by elros34**
… for authentication

### CODE REVIEW

**Comment by winterz** [comment]
We really appreciate the contributions you are making to KDSoap.

Since KDSoap is dual-licensed as a commercial KDAB offering we need you
to fill-in, scan and return the Copyright Assignment form (see doc/KDSoap-CopyrightAssignmentForm.docx)
You can also snail mail to our KDAB offices, if you prefer.

If you are doing this work for your employer there is a special section to cover that.

Let me know if you have any questions.
-Allen

**Comment by elros34** [comment]
User name and password are public. I use kdsoap to track parcels in Polish Postal Service:
https://www.poczta-polska.pl/webservices/
Is it really necessary to fill some kind of document with my personal data for modification which basically adds one simple boolean check?

File: kdsoap.pro [hunk 1]
```diff
@@ -106,7 +106,7 @@ QMAKE_EXTRA_TARGETS += test
 
 INSTALL_DOC_DIR = $$INSTALL_PREFIX/share/doc/KDSoap
 # install licenses:
-licenses.files = LICENSE.GPL.txt LICENSE.txt
+licenses.files = LICENSE.LGPL.txt LICENSE.GPL.txt LICENSE.txt
```
Comment (martonmiklos) [inline kdsoap.pro:109]: I think it is an accidental change.

File: src/KDSoapClient/KDSoapAuthentication.h [hunk 2]
```diff
@@ -78,6 +78,15 @@ class KDSOAP_EXPORT KDSoapAuthentication
      */
     QString password() const;
 
+    /**
+     * Sets whether #PasswordDigest (default) or #PasswordText is used for authentication
+     */
+    void setUsePasswordDigest(const bool usePasswordDigest);
```
Comment (dfaure-kdab) [inline src/KDSoapClient/KDSoapAuthentication.h:84]: (the const here serves no purpose, unlike in the .cpp file)

**Pull request merged**